lib/main.dart

```diff
@@ -17,10 +18,12 @@ class MyApp extends StatelessWidget {
   Widget build(BuildContext context) {
     final appRouter = AppRouter();
 
-    return MultiProvider( // Wrap your MaterialApp.router with MultiProvider
+    return MultiProvider(
+      // Wrap your MaterialApp.router with MultiProvider
       providers: [
-        ChangeNotifierProvider(create: (context) => SearchScreenProvider()), // Provide the SearchScreenProvider
-        // Add more providers if needed
+        ChangeNotifierProvider(create: (context) => SearchScreenProvider()),
+        ChangeNotifierProvider(
```

Comment:
Do you need to create ProductDetailsScreenProvider at this point of the app? Seems like it can be moved to the ProductDetails page

Reply:
No, it won't work as it is used in the home screen to update variables before entering the screen

Reply:
What variables need to be updated before we enter the screen?